client/my-sites/plugins/plugins-dashboard/index.tsx

```diff
@@ -115,8 +116,8 @@ const PluginsDashboard = ( {
 	const productsList = useSelector( ( state ) => getProductsList( state ) );
 	const { data: dotComPlugins }: { data: Plugin[] | undefined } = useWPCOMPluginsList( 'all' );
 	const allPlugins = useSelector( ( state ) =>
-		getPlugins( state, siteIds, 'all' ).map( ( plugin: Plugin ) => {
-			let dotComPluginData: Plugin | undefined;
+		getPlugins( state, siteIds, 'all' ).map( ( plugin: PluginComponentProps ) => {
```

Comment:
Using `PluginComponentProps` as a type as it contains the `isMarketplaceProduct` property and it does not need to be added to the `Plugin` type below.